malcolm/modules/ADPandABlocks/parts/kinematicssavupart.py

```diff
@@ -0,0 +1,353 @@
+                q_value = pmac.util.CS_AXIS_NAMES.index(mapping.cs_axis) + 1
+                self.q_value_mapping[q_value] = mapping.scannable
+
+        assert "." in self.nxs_full_filename, \
+            "File extension for %r should be supplied" % self.nxs_full_filename
+
+    @add_call_types
+    def post_configure(self, context, part_info):
+        # type: (scanning.hooks.AContext, scanning.hooks.APartInfo) -> None
+
+        self.pos_table = context.block_view(self.panda_mri).positions.value
+
+        # Get the axis number for the inverse kinematics mapped in this cs_port
+        self.axis_numbers = pmac.util.cs_axis_numbers(
+            context, self.layout_table, self.cs_port
+        )
+
+        # Map these in the file
+        dataset_infos = scanning.infos.DatasetProducedInfo.filter_values(
+            part_info
+        )
+        for scannable, axis_num in self.axis_numbers.items():
+            min_i, max_i, value_i = None, None, None
+            for info in dataset_infos:
+                if info.name.startswith(scannable + "."):
+                    if info.type == scanning.infos.DatasetType.POSITION_MIN:
+                        min_i = info
+                    elif info.type == scanning.infos.DatasetType.POSITION_MAX:
+                        max_i = info
+                    elif info.type == scanning.infos.DatasetType.POSITION_VALUE:
+                        value_i = info
+            # Always make sure .value is there
+            assert value_i, "No value dataset for %s" % scannable
+            self.p_vars.append(PVar(
+                path=value_i.path, file=value_i.filename,
+                p_number="p%dmean" % axis_num)
+            )
+            if min_i and max_i:
+                self.p_vars.append(
+                    PVar(
+                        path=min_i.path, file=min_i.filename,
+                        p_number="p%dmin" % axis_num
+                    )
+                )
+                self.p_vars.append(PVar(path=max_i.path, file=max_i.filename,
+                                        p_number="p%dmax" % axis_num))
+            else:
+                self.use_min_max = False
+
+        # Get Forward Kinematics code lines and I,P,M,Q input variables
+        pmac_status_child = context.block_view(self.pmac_mri + ":STATUS")
+
+        raw_input_vars = " ".join([pmac_status_child.iVariables.value,
+                                   pmac_status_child.pVariables.value,
+                                   pmac_status_child.mVariables.value])
+
+        pmac_cs_child = context.block_view(
+            self.pmac_mri + ":" + self.cs_mri_suffix)
+
+        raw_kinematics_program_code = pmac_cs_child.forwardKinematic.value
+        raw_input_vars += " " + pmac_cs_child.qVariables.value
+
+        self.savu_code_lines = raw_kinematics_program_code.splitlines()
+        self.parse_input_variables(raw_input_vars)
+
+        self.create_files()
+
```

Comment:
We should return Infos for the datasets we're going to produce here (the calculated ones)

Reply:
Should it be configure or post_configure which returns the info?

Reply:
on_configure. Liker here: https://github.com/dls-controls/pymalcolm/blob/master/malcolm/modules/ADOdin/parts/odinwriterpart.py#L340

Reply:
Currently the datasets produced aren't determined until post_configure, since it checks if the PandA is going to produce Min/Max/Mean or just value and produces the same. Is there an easy way around this?

Reply:
Unfortunately the datasets Table is not populated by on_configure. Fortunately all the information is the panda positions Table, which you are already reading to get mres and offset. You could move some of that code to on_configure, then produce the dataset infos there too...

Reply:
Do we need to consider the possibility of configure loading a different design (with a different pos table) on the PandA, or does this happen before on_configure is called/is it safe to ignore this?

Reply:
This now happens at PreConfigure, so when on_configure is called the correct design will already be loaded